news/2021-11-08-performance-points-star-rating-updates.md

```diff
@@ -0,0 +1,132 @@
+
+### New Flashlight difficulty skill
+
+Flashlight in difficulty calculation is now revamped, as it moves from an object count-based multiplier applied to the Aim skill to a brand new skill of its own. 
+
+The difficulty that Flashlight adds to a map is now measured much more accurately, thanks to [**MBmasher**](https://osu.ppy.sh/users/4498616)'s change in [this pull request](https://github.com/ppy/osu/pull/14217).
+
+Previously, the object count-based method posed a number of issues. Flow aim (used for hitting streams) is mostly represented by the speed skill, not aim. Additionally, the aim skill, unsurprisingly, only measures the aim difficulty, not the Flashlight difficulty. To put this simply, there will be types of jumps that are hard to aim but quite easy to read with Flashlight, and vice versa.
+
+For example, let's consider a familiar type of jump pattern: a back and forth. This is an example of a jump that can have high aim difficulty, but relatively low Flashlight difficulty. The distance between the first object and the third object in a back and forth is typically small enough for it to be within the Flashlight radius, thus allowing a player to read the position of that circle very easily.
+
+![](/wiki/shared/news/2021-11-08-performance-points-star-rating-updates/visible-fl-area.png)
+
+The creation of a Flashlight skill makes it possible to have it contribute to star rating. [This has been enabled](https://github.com/ppy/osu/pull/14753) following the results of the recent osu! performance points and star rating survey, which revealed that players desire star rating to be a metric of difficulty, and that they want subjective skills included. Ask and thou shall receive!
+
+It is also worth noting that the Hidden mod now applies a 1.3x multiplier on the Flashlight skill. The visibility of hit objects are decreased significantly due to the fading of circles and the absence of approach circles.
+
+Find yourself curious about the details? [MBmasher](https://osu.ppy.sh/users/4498616)'s write-up about the implementation will very likely scratch that itch, so feel free to [**indulge in some Flashlight theory**](https://docs.google.com/document/d/11FBEEPXK8yI2d5uTZuebJC3S9nVUKgMD8WdvblERyYA/edit).
+
+### Speed hard cap removal
+
+The speed hard cap in osu! has been removed by [**emu1337**](https://osu.ppy.sh/users/2185987) and [**Apo11o**](https://osu.ppy.sh/users/9558549) in [this pull request](https://github.com/ppy/osu/pull/14617), allowing 300+ BPM streams to be weighed accurately. 
+
+The speed cap was previously set to 300 BPM to prevent abuse cases, which means that calculating difficulty on beatmaps with higher BPM requires special consideration. Two adjustments were made to account for that:
+
+- Speed is nerfed on doubles that are double-tappable. If the time since the last object for the current object is smaller than the hit window for a 300, we instead take the difference between that object and the hit window of the last one.
```

Comment:
Ah, no, I have no idea what this wording is trying to convey.

[The change in question for reference](https://github.com/ppy/osu/pull/14617/files#diff-2877a1e4555d4b301a40010c0693e8a8ce3f9bcd8759f1bc22b887d62a03b459R55-R56):

```cs
double speedWindowRatio = strainTime / greatWindowFull;

// Aim to nerf cheesy rhythms (Very fast consecutive doubles with large deltatimes between)
if (osuPrevious != null && strainTime < greatWindowFull && osuPrevious.StrainTime > strainTime)
                strainTime = Interpolation.Lerp(osuPrevious.StrainTime, strainTime, speedWindowRatio);
```

By inspection of the code, for a tight double, the time difference between its notes will be considered as if it was closer to the time difference of the gap before the double.

For a double with notes 25 ms apart (the minimum in calculations) and hit window of 50 ms (about AR 9.6 I think), the ratio of this weighted sum would be 0.5. For lower AR, doubles are nerfed harder and vice versa.

```suggestion
- Speed is nerfed on doubles that are double-tappable. If the time gap between two notes in a double is smaller than the hit window for a 300, the speed skill calculation will consider it as if it was closer to the time gap before the double.
```

... or whatever you find most illustrative.

Reply:
The above seems fine, but I'd probably just omit/skip over the interpolation details, mostly. I'm not sure it's worth the attention. I'd probably leave it at something like

```suggestion
- A double can be considered double-tappable if the time interval between both notes of the double is small enough that their hit windows overlap, which makes it harder to miss both when a key is pressed twice in the overlap period. Those sorts of patterns are nerfed by artificially increasing the time gap between them.
```

Reply:
I can agree with something like this, but I want to turn it around to front-load the information

```suggestion
Quick doubles with overlapping hit windows are nerfed. With a small enough time interval between two notes, they can still be perfectly hit by simply double-tapping during the overlap of their hit windows. This is now accounted for in calculations by artificially increasing the time gap between the notes.
```